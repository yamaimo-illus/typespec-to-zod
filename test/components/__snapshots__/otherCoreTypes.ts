import { z } from "zod";

export const schemaOtherCoreTypes = z.object({
    bytesProperty: z.string(),
    stringProperty: z.string().default("default value"),
    booleanProperty: z.boolean().default(true),
    nullProperty: z.unknown(),
    ArrayProperty: z.array(z.number()).default([0, 1, 2]),
    unknownProperty: z.unknown()
});