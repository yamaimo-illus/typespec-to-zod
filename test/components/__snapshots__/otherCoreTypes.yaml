openapi: 3.0.0
info:
  title: (title)
  version: 0.0.0
tags: []
paths: {}
components:
  schemas:
    OtherCoreTypes:
      type: object
      required:
        - bytesProperty
        - stringProperty
        - booleanProperty
        - nullProperty
        - ArrayProperty
        - unknownProperty
      properties:
        bytesProperty:
          type: string
          format: byte
        stringProperty:
          type: string
          default: default value
        booleanProperty:
          type: boolean
          default: true
        nullProperty:
          nullable: true
        ArrayProperty:
          type: array
          items:
            type: number
          default:
            - 0
            - 1
            - 2
        unknownProperty: {}
